Mathlib/Topology/Algebra/Algebra.lean

```diff
@@ -74,16 +78,471 @@ theorem algebraMapCLM_toLinearMap : (algebraMapCLM R A).toLinearMap = Algebra.li
+    ContinuousAlgHomClass (A →A[R] B) R A B where
+  map_mul f x y    := map_mul f.toAlgHom x y
+  map_one f        := map_one f.toAlgHom
+  map_add f        := map_add f.toAlgHom
+  map_zero f       := map_zero f.toAlgHom
+  commutes f r     := f.toAlgHom.commutes r
+  map_continuous f := f.2
+
+theorem coe_mk (f : A →ₐ[R] B) (h) : (mk f h : A →ₐ[R] B) = f := rfl
+
+@[simp]
+theorem coe_mk' (f : A →ₐ[R] B) (h) : (mk f h : A → B) = f := rfl
+
+@[continuity]
+protected theorem continuous (f : A →A[R] B) : Continuous f := f.2
+
+protected theorem uniformContinuous {E₁ E₂ : Type*} [UniformSpace E₁] [UniformSpace E₂]
+    [Ring E₁] [Ring E₂] [Algebra R E₁] [Algebra R E₂] [UniformAddGroup E₁]
+    [UniformAddGroup E₂] (f : E₁ →A[R] E₂) : UniformContinuous f :=
+  uniformContinuous_addMonoidHom_of_continuous f.continuous
+
+@[simp, norm_cast]
+theorem coe_inj {f g : A →A[R] B} : (f : A →ₐ[R] B) = g ↔ f = g := coe_injective.eq_iff
+
+theorem coeFn_injective : @Function.Injective (A →A[R] B) (A → B) (↑) := DFunLike.coe_injective
+
+/-- See Note [custom simps projection]. We need to specify this projection explicitly in this case,
+  because it is a composition of multiple projections. -/
+def Simps.apply (h : A →A[R] B) : A → B := h
+
+/-- See Note [custom simps projection]. -/
+def Simps.coe (h : A →A[R] B) : A →ₐ[R] B := h
+
+initialize_simps_projections ContinuousAlgHom (toAlgHom_toFun → apply, toAlgHom → coe)
+
+@[ext]
+theorem ext {f g : A →A[R] B} (h : ∀ x, f x = g x) : f = g := DFunLike.ext f g h
+
+theorem ext_iff {f g : A →A[R] B} : f = g ↔ ∀ x, f x = g x := DFunLike.ext_iff
+
+/-- Copy of a `ContinuousAlgHom` with a new `toFun` equal to the old one. Useful to fix
+definitional equalities. -/
+def copy (f : A →A[R] B) (f' : A → B) (h : f' = ⇑f) : A →A[R] B where
+  toAlgHom := {
+    toRingHom := (f : A →A[R] B).toRingHom.copy f' h
+    commutes' := fun r => by
+      simp only [AlgHom.toRingHom_eq_coe, h, RingHom.toMonoidHom_eq_coe, OneHom.toFun_eq_coe,
+        MonoidHom.toOneHom_coe, MonoidHom.coe_coe, RingHom.coe_copy, AlgHomClass.commutes f r] }
+  cont := show Continuous f' from h.symm ▸ f.continuous
+
+@[simp]
+theorem coe_copy (f : A →A[R] B) (f' : A → B) (h : f' = ⇑f) : ⇑(f.copy f' h) = f' := rfl
+
+theorem copy_eq (f : A →A[R] B) (f' : A → B) (h : f' = ⇑f) : f.copy f' h = f := DFunLike.ext' h
+
+protected theorem map_zero (f : A →A[R] B) : f (0 : A) = 0 := map_zero f
+
+protected theorem map_add (f : A →A[R] B) (x y : A) : f (x + y) = f x + f y := map_add f x y
+
+protected theorem map_smul (f : A →A[R] B) (c : R) (x : A) :
+    f (c • x) = c • f x := map_smul _ _ _
+
+theorem map_smul_of_tower {R S : Type*} [CommSemiring S] [SMul R A] [Algebra S A] [SMul R B]
+    [Algebra S B] [MulActionHomClass (A →A[S] B) R A B] (f : A →A[S] B) (c : R) (x : A) :
+    f (c • x) = c • f x :=
+  map_smul f c x
+
+@[deprecated _root_.map_sum (since := "2023-09-16")]
+protected theorem map_sum {ι : Type*} (f : A →A[R] B) (s : Finset ι) (g : ι → A) :
```

Comment:
Why are you adding a deprecated lemma? Also, the date is wrong. Let's just exclude it entirely.